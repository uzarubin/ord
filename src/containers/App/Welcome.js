/*
  @flow
*/

import React from 'react'

import { Text } from 'react-native'

type Props = {
  msg: string,
}

export default function (props: Props) {
  return <Text> {props.msg ? props.msg : 'Welcome to React Native'} </Text>
}
