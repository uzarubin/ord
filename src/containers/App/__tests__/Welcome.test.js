import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import Welcome from '../Welcome'

// Snapshot testing
describe('Testing Welcome text component', () => {
  it('renders default message without msg prop', () => {
    const tree = renderer.create(<Welcome />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders msg when msg prop is passed', () => {
    const wrapper = shallow(<Welcome msg="Hello from tests!" />)
    expect(wrapper).toMatchSnapshot()
    wrapper.setProps({ msg: 'Hello World from tests!' })
    expect(wrapper).toMatchSnapshot()
  })
})
