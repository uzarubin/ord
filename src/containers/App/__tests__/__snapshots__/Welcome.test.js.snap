// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Welcome text component renders default message without msg prop 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  Welcome to React Native
</Text>
`;

exports[`Testing Welcome text component renders msg when msg prop is passed 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  Hello from tests!
</Text>
`;

exports[`Testing Welcome text component renders msg when msg prop is passed 2`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  Hello World from tests!
</Text>
`;
